### ISSUE TITLE
Multiple calls to reserveCredits results in violation of transactions_pkey constraint

### ISSUE THREAD

**Issue by xiroV** [SDU-eScience/UCloud#2371]
When `reserveCredits` is called for a job which already exists, it results in a violation of the `transactions_pkey` constraint.

```
com.github.jasync.sql.db.postgresql.exceptions.GenericDatabaseException: ErrorMessage(fields=[(Severity, ERROR), (V, ERROR), (SQLSTATE, 23505), (Message, duplicate key value violates unique constraint "transactions_pkey"), ...
```

This happens because we attempt to insert a transaction with the same `jobId`, `accountId` and `accountType`, which already exists.

According to @DanThrane ambassador sometimes refuses to forward requests to other services, leading to multiple calls, which could lead to the multiple calls to `reserveCredits` as well.

**Comment by xiroV** [comment]
Because of the redesign on the `storage` branch, this issue is most likely not a problem anymore.

**Comment by DanThrane** [comment]
Fixed as of 2021.3.0
